modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -502,6 +502,31 @@ func (h *Handler) proxyLoopIteration(r *http.Request, origReq *http.Request, w h
 	// or satisfactorily represented in a URL
 	caddyhttp.SetVar(r.Context(), dialInfoVarKey, dialInfo)
 
+	var proxyProtocolInfo ProxyProtocolInfo
+	// using X-Forwarded-For header which is already filtered by trusted proxies
+	if xff := r.Header.Get("X-Forwarded-For"); xff != "" {
+		ip := net.ParseIP(xff)
```

Comment:
This isn't safe, `X-Forwarded-For` could contain a comma separated list of IP addresses.

Reply:
I forget that caddy set it as a string 😇, will change it.